### PR TITLE
Changes to make pidfile work on Windows

### LINKED ISSUE #1992: Duplicate instance checks are abstracted per-platform

### PR DESCRIPTION
**Pull request by yying**
Closes #1992

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: osquery/core/process.h [hunk 1]
```diff
@@ -66,6 +66,10 @@ class PlatformProcess : private boost::noncopyable {
   explicit PlatformProcess() : id_(kInvalidPid) {}
   explicit PlatformProcess(PlatformPidType id);
 
+#ifdef WIN32
+  explicit PlatformProcess(pid_t pid);
```
Comment (theopolis) [inline osquery/core/process.h]: Mind adding a note, or docstring, to why `PlatformPidType` is not a `pid_t` on both platforms?

**Comment by theopolis** [comment]
ok to test

**Comment by osqueryer** [comment]
Build origin/pr/2297/merge (Job: 2945) failed :-1:

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Build origin/pr/2297/merge (Job: 2948) failed :-1:

**Comment by osqueryer** [comment]
Build origin/pr/2297/merge (Job: 2949) is good! :+1:

**Comment by osqueryer** [comment]
Build origin/pr/2297/merge (Job: 2950) failed :-1:

File: osquery/core/process.h [hunk 2]
```diff
@@ -130,6 +143,30 @@ class PlatformProcess : private boost::noncopyable {
   PlatformPidType id_;
 };
 
+#ifdef WIN32
+/// Class to handle the scope of a SecurityDescriptor from
+/// GetSecurityInfo/GetNamedSecurityInfo class of functions
```
Comment (theopolis) [inline osquery/core/process.h]: nit, use multi-line docstring syntax.

File: osquery/core/system.cpp [hunk 3]
```diff
@@ -187,45 +188,54 @@ Status checkStalePid(const std::string& content) {
...
-        status = kill(pid, SIGKILL);
-        ::sleep(1);
+  // The pid is running, check if it is an osqueryd process by name.
+  std::stringstream query_text;
 
-        return Status(status, "Tried to force remove the existing osqueryd");
-      }
+#ifdef WIN32
+  query_text << "SELECT name FROM processes WHERE pid = " << pid
+             << " AND name = 'osqueryd.exe';";
```
Comment (theopolis) [inline osquery/core/system.cpp]: Wait, why not `AND name LIKE 'osqueryd%'`

File: osquery/core/system.cpp [hunk 4]
```diff
@@ -187,45 +188,54 @@ Status checkStalePid(const std::string& content) {
...
   return Status(0, "OK");
 }
 
 Status createPidFile() {
   // check if pidfile exists
-  auto exists = pathExists(FLAGS_pidfile);
+  auto pidfile_path = fs::path(FLAGS_pidfile).make_preferred();
+
+  auto exists = pathExists(pidfile_path);
   if (exists.ok()) {
```
Comment (theopolis) [inline osquery/core/system.cpp]: If the `exists` status is not used to emit messages/log, you can just: 
```cpp
if (pathExists(pidfile_path).ok()) {
```

File: osquery/filesystem/filesystem.cpp [hunk 5]
```diff
@@ -77,14 +77,18 @@ Status writeTextFile(const fs::path& path,
 
 struct OpenReadableFile {
  public:
-  explicit OpenReadableFile(const fs::path& path) {
+  explicit OpenReadableFile(const fs::path& path, bool is_nonblocking = true) {
```
Comment (theopolis) [inline osquery/filesystem/filesystem.cpp]: Q: is there precedent for using `is_`? Can we name this `bool blocking = false`?

File: osquery/core/system.cpp (hunk 3 above)
Comment (yying) [inline osquery/core/system.cpp]: I was being specific but I'm fine with either approach.

**Comment by osqueryer** [comment]
Build origin/pr/2297/merge (Job: 2953) is good! :+1:

**Pull request merged**